src/main/kotlin/org/rust/cargo/project/configurable/RsProjectConfigurable.kt

```diff
@@ -42,34 +46,36 @@ class RsProjectConfigurable(
             checkBox(RsBundle.message("settings.rust.toolchain.inject.rust.in.doc.comments.checkbox"))
                 .bindSelected(state::doctestInjectionEnabled)
         }
+
+        onApply {
+            settings.modify {
+                it.toolchain = rustProjectSettings.data.toolchain
+                it.explicitPathToStdlib = rustProjectSettings.data.explicitPathToStdlib
+                it.macroExpansionEngine = state.macroExpansionEngine
+                it.doctestInjectionEnabled = state.doctestInjectionEnabled
+            }
+        }
+        onReset {
+            rustProjectSettings.data = RustProjectSettingsPanel.Data(
+                toolchain = settings.toolchain ?: RsToolchainBase.suggest(projectDir),
+                explicitPathToStdlib = settings.explicitPathToStdlib
+            )
```

Comment:
Let's do it only if `rustProjectSettings.data` is changed. Otherwise, the corresponding field blinks on each reset

https://user-images.githubusercontent.com/2539310/234810385-4c4bacaa-98e7-471e-96c5-6d1aa1b1a512.mov

I understand that it's current behavior but why don't fix it?